dds_cli/data_putter.py

```diff
@@ -169,6 +170,76 @@ def put(
                         for x in [y.id for y in progress.tasks if y.fields.get("step") != "put"]
                     ]
 
+        # LOG.warning(f"test 1: {putter.filehandler.failed}")
+        if pathlib.Path(putter.failed_delivery_log).is_file():
+            LOG.warning(
+                "Some uploaded files could not be added to the database %s.\n",
+                str(putter.failed_delivery_log),
+            )
+            LOG.warning("Attempting to add them to the database.")
+
+            path_to_file = str(putter.failed_delivery_log)
+            with open(path_to_file, "r", encoding="utf-8") as json_f:
+                failed = json.load(json_f)
+
+            # remove from log any files that  failed for other reasons
+            for file, values in failed.items():
+                if values.get("status", {}).get("failed_op") != "add_file_db":
+                    failed.pop(file)
+
+            # send failed to API endpoint
+            params = {"project": putter.project}
+            try:
+                response, _ = dds_cli.utils.perform_request(
+                    DDSEndpoint.FILE_UPDATE_FAILED,
+                    method="put",
+                    headers=putter.token,
+                    params=params,
+                    json=failed,
+                    error_message="Failed to add missing files",
+                )
+            except (
+                dds_cli.exceptions.ApiRequestError,
+                dds_cli.exceptions.ApiResponseError,
+                dds_cli.exceptions.DDSCLIException,
+            ) as err:
+                message = str(err)
+                LOG.warning(message)
+
+            message = response.get("message")
+            if not message:
+                raise dds_cli.exceptions.ApiResponseError(
+                    message="Failed to add missing files to database."
+                )
```

Comment:
Feels like a duplicate of the code handling the request, don't think we need it

Reply:
I guess you mean duplicate of line 203, don't you!

Reply:
Actually forget what I said, keep this